packages/legend-query-builder/src/stores/explorer/QueryFunctionsExplorerState.ts

```diff
@@ -347,25 +405,83 @@ export class QueryFunctionsExplorerState {
     }
   };
 
+  initializeFunctionInfoMap(): void {
+    const functionInfoMap = new Map<string, FunctionAnalysisInfo>();
+    const dependencyFunctionInfoMap = new Map<string, FunctionAnalysisInfo>();
+    const functionInfos =
+      buildFunctionAnalysisInfoFromConcreteFunctionDefinition(
+        this.queryBuilderState.graphManagerState.graph.ownFunctions,
+        this.queryBuilderState.graphManagerState.graph,
+      );
+    functionInfos.forEach((info) =>
+      functionInfoMap.set(info.functionPath, info),
+    );
+    if (
+      this.queryBuilderState.graphManagerState.graph.dependencyManager
+        .hasDependencies
+    ) {
+      const dependencyFunctions =
+        this.queryBuilderState.graphManagerState.graph.dependencyManager
+          .functions;
+      const dependencyFunctionInfos =
+        buildFunctionAnalysisInfoFromConcreteFunctionDefinition(
+          dependencyFunctions,
+          this.queryBuilderState.graphManagerState.graph,
+        );
+      dependencyFunctionInfos.forEach((info) =>
+        dependencyFunctionInfoMap.set(info.functionPath, info),
+      );
+    }
+    const queryBuilderFunctionAnalysisInfo =
+      this.queryBuilderState.getExtraQueryBuilderFunctionAnalysisInfo();
+    if (queryBuilderFunctionAnalysisInfo) {
+      Array.from(
+        queryBuilderFunctionAnalysisInfo.functionInfoMap.entries(),
+      ).forEach(([path, info]) => functionInfoMap.set(path, info));
+      Array.from(
+        queryBuilderFunctionAnalysisInfo.dependencyFunctionInfoMap.entries(),
+      ).forEach(([path, info]) => {
+        dependencyFunctionInfoMap.set(path, info);
+      });
+    }
+    const packagePathToFunctionInfoMap = new Map<
+      string,
+      FunctionAnalysisInfo[]
+    >();
+    Array.from(functionInfoMap.values())
+      .concat(Array.from(dependencyFunctionInfoMap.values()))
+      .forEach((info) => {
+        const curr = packagePathToFunctionInfoMap.get(info.packagePath);
+        if (curr) {
+          packagePathToFunctionInfoMap.set(info.packagePath, [...curr, info]);
+        } else {
+          packagePathToFunctionInfoMap.set(info.packagePath, [info]);
+        }
+      });
+    this.setPackagePathToFunctionInfoMap(packagePathToFunctionInfoMap);
+    this.setFunctionInfoMap(functionInfoMap);
+    this.setDependencyFunctionInfoMap(dependencyFunctionInfoMap);
+  }
+
   initializeTreeData(): void {
```

Comment:
could we move this to be called only if the function explorer panel has been opened. should only happen if they trigger it to be opened.